Automattic/jetpack
# Notifications and Reader: Fix Spotify shortcode render

#### Changes proposed in this Pull Request:
This commit addresses the issue where the Spotify shortcode isn't properly rendered in WPCOM notifications and Reader.

The proposed code checks whether it is running on WPCOM and then confirms the display context with the help of `/lib/display-context.php`.

Reader render: **before**
![Markup_on_2021-08-18_at_14_06_31](https://user-images.githubusercontent.com/25105483/131468016-18664951-bba2-44d4-b87b-71b224e8c9b6.png)

Reader render: **after**
![Markup_on_2021-08-18_at_14_10_25](https://user-images.githubusercontent.com/25105483/131468021-47a4cb92-d7ee-4940-9952-e0c7dc53bbfd.png)

Notification render: **before**
![Markup_on_2021-08-18_at_14_08_00](https://user-images.githubusercontent.com/25105483/131468177-04d8376c-e1e3-41d3-bdfd-ee6f0529a322.png)

Notification render: **after**
![Markup_on_2021-08-18_at_14_09_36](https://user-images.githubusercontent.com/25105483/131468216-74ea2c06-4cda-4981-82f5-ec8d5c5fd960.png)

#### Does this pull request change what data or activity we track or use?
No, it does not.

#### Testing instructions:
1. Apply the generated patch to WPCOM sandbox as it is outlined in p7H4VZ-30i-p2.
2. Make sure there is a subscriber on your site that can receive notifications on newly published posts.
Notification settings are located in the subscriber's account at `/following/manage` -> (gear icon) "Settings" -> "Notify me of new posts" (toggle):
![Markup_on_2021-07-29_at_15_35_56](https://user-images.githubusercontent.com/25105483/131467803-d4732578-a3b5-4a25-a061-7a1c7fd15da6.png)
3. Log in to the account of the user that can publish posts on the site.
4. Create and publish a new post that has the `[spotify]` shortcode in it. Example shortcode that can be used: `[spotify https://open.spotify.com/playlist/5PKZSKuHP4d27SXO5fB9Wl]`.
5. Review the notification in the subscriber's account.
6. Review the post in the Reader.
7. The shortcode should be rendered correctly in both - notification and Reader - similar to the screenshots above.
8. The link in the notification should open in a new tab.
